util/parsefiles.py

```
@@ -0,0 +1,280 @@
+        self._map_plays_to_roles(graph, docker_play_dirs, git_dir, "roles", "docker_playbook", "role")
+
+        return graph
+
+    def _map_roles_to_roles(self, graph, dirs, git_dir, key, type_1, type_2):
+        """Maps roles to the roles that they depend on."""
+
+        Node = namedtuple('Node', ['name', 'type'])
+
+        # for each role directory
+        for d in dirs:
+            d = pathlib2.Path(git_dir, d)
+
+            if d.is_dir():
+                # for all files/sub-directories in directory
+                for directory in d.iterdir():
+
+                    # attempts to find meta/*.yml file in directory
+                    role = [file for file in directory.glob("meta/*.yml")]
+
+                    # if role exists
+                    if role:
+                        with (open(str(role[0]), "r")) as file:
+                            yaml_file = yaml.load(file)
+
+                        # if a yaml file and key in file
+                        if yaml_file is not None and key in yaml_file:
+                            # for each dependent role
+                            for dependent in yaml_file[key]:
+                                # get role name
+                                name = self._get_role_name(dependent)
+
+                                # add node for role
+                                node_1 = Node(directory.name, type_1)
+                                # add node for dependent role
+                                node_2 = Node(name, type_2)
+                                # add edge role - dependent role
+                                graph.add_edge(node_1, node_2)
+
+    def _map_plays_to_roles(self, graph, dirs, git_dir, key, type_1, type_2):
+        """Maps plays to the roles they use."""
+
+        Node = namedtuple('Node', ['name', 'type'])
+
+        # for each play directory
+        for d in dirs:
+            d = pathlib2.Path(git_dir, d)
+
+            if d.is_dir():
```

If a playbooks dir doesn't exist, this tool should fail noisily, so I don't there's any need for this check.